Sources/Testing/Running/Configuration.TestFilter.swift

```diff
@@ -422,3 +485,60 @@ extension Configuration.TestFilter {
     self = combining(with: other, using: op)
   }
 }
+
+// MARK: - Filterable types
+
+extension Configuration.TestFilter.Kind {
+  /// Whether this kind of test filter requires knowledge of test traits.
+  ///
+  /// If the value of this property is `true`, the values of a test graph must
+  /// be converted to `FilterItem` and have trait information recursively
+  /// propagated before the filter can be applied, or else the results may be
+  /// inaccurate. This facilitates a performance optimization where trait
+  /// propagation can be skipped for filters which don't require such knowledge.
+  fileprivate var requiresFilterItemConversion: Bool {
```

Comment:
I don't have a better name for this property, but I wonder if we can come up with one that doesn't just describe its own technical effects. `requiresRecursiveBlahBlah` or somethin'.

(Not critical for this PR, but worth bikeshedding if we think this property or pattern is going to be useful in other places too.)

Reply:
Renamed to `requiresTraitPropagation`. Not going to sweat this one too much further though, especially since it's `fileprivate`